### ISSUE TITLE
`gather` function fails after padding faces with `-1` for variable length meshes

### ISSUE THREAD

**Issue by qixuema** [lucidrains/meshgpt-pytorch#5]
Hello,

I've encountered an issue with the [`gather` function](https://github.com/lucidrains/meshgpt-pytorch/blob/d8b12f2514013bb9265686ca1caf1ace6ceb9577/meshgpt_pytorch/meshgpt_pytorch.py#L364C32-L364C38) after padding operation. As suggested in the usage notes:

> "# make sure faces and face_edges are padded with `-1` for variable lengthed meshes. otherwise, you will need to explicitly pass in `face_len` as well as `face_edges_len`."

Following this, I padded the `faces` tensor with `-1` to account for variable length meshes. However, this seems to have caused the `gather` function to fail. The padding was done by setting the values of the last n faces in `faces` entirely to `-1`.

Is the padding process causing the `gather` operation to fail? Or is there a specific way the padding should be performed in this context? Below is the snippet of my code where the padding is applied:

```python
import torch
import random

t = torch.randint(0, 121, (2, 64, 2))
for i in range(t.shape[0]):
    n = random.randint(1, t.shape[1])    
    t[i, -n:, :] = -1
```

**Comment by lucidrains** [comment]
@qixuema ah, i didn't apply the face mask at the right time

want to check out 0.0.52?

**Comment by qixuema** [comment]
Thank you very much for the swift response and resolution to the issue with the `gather` function following padding with `-1`. I appreciate the time and effort you took to address the problem so quickly. 

The solution works perfectly and my code is now running as expected. I'm grateful for your support and the excellent work you're doing with this project. Thanks again!

**Comment by lucidrains** [comment]
@qixuema no problem Xueqi! thank you for reporting this error
